src/main/java/ui/MyStage.java

```diff
@@ -0,0 +1,191 @@
+        return sb.append(SEPARATOR).append("\n").append(message).append(SEPARATOR).toString();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    private String getResponse(String userCommand) {
+        try {
+            Command newCommand = commandParser.parse(userCommand);
+            newCommand.parse();
+            DukeResponse response = newCommand.execute(service);
+            return printMessage(response.toString());
+        } catch (Exception e) {
+            return e.toString();
+        }
+    }
+
+    /**
+     * Constructs a new MyStage
+     * @param service a DukeService which executes commands
+     * @param commandParser a CommandParser to parse commands
+     * @param taskParser a TaskParser to parse tasks
+     * @param storage a Storage to store and retrieve data from text files.
+     */
+    public MyStage(DukeService service, CommandParser commandParser, TaskParser taskParser, Storage storage) {
+        this.service = service;
+        this.commandParser = commandParser;
+        this.taskParser = taskParser;
+        this.storage = storage;
+    }
+
+    public void setImage(Image user, Image duke) {
+        this.user = user;
+        this.duke = duke;
+    }
+
+
+    private void printGreeting() {
+        String text = "Hello!";
+        Label dukeText = new Label(text);
+        dialogContainer.getChildren().addAll(
+                DialogBox.getDukeDialog(dukeText, new ImageView(duke))
+        );
+    }
+
+    /**
+     * Iteration 2:
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    private void handleUserInput() {
+        String userCommand = userInput.getText();
+
+        Label userText = new Label(userCommand);
+        Label dukeText = new Label(getResponse(userCommand));
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(userText, new ImageView(user)),
+                DialogBox.getDukeDialog(dukeText, new ImageView(duke))
+        );
+        userInput.clear();
+        if (userCommand.equals("bye")) {
+            try {
+                storage.writeToFile(service);
+            } catch (DukeException ignored) {
+                printMessage(ignored.getMessage());
+            }
+            Platform.exit();
+            System.exit(0);
+        }
+
+    }
+
+    /**
+     * Starts the show!
+     * @param stage the stage of the show
+     */
+    public void start(Stage stage) {
+        //Step 1. Setting up required components
+
+        //The container for the content of the chat to scroll.
+        scrollPane = new ScrollPane();
+        dialogContainer = new VBox();
+        scrollPane.setContent(dialogContainer);
+
+        userInput = new TextField();
+        sendButton = new Button("Send");
```

Comment:
Maybe you could try avoid "magic" number/string like "Send", 600.0, 400.0? Maybe can consider adding named constants for them?